service/feeder/ledger_wal.go

```diff
@@ -100,34 +113,56 @@ func (l *LedgerWAL) Delta(commit flow.StateCommitment) (dps.Delta, error) {
 			return nil, dps.ErrFinished
 		}
 		record := l.reader.Record()
-		operation, _, update, err := wal.Decode(record)
+		duplicate := make([]byte, len(record))
+		copy(duplicate, record)
```

Comment:
I'd suggest including a comment on why we need to do this like you explained earlier. It might become a puzzling discovery for a fellow developer in the future otherwise, especially since it might be confused with the deduplication code just under, which is completely unrelated.